src/components/audio-ext-element.js

```diff
@@ -58,67 +58,69 @@ export default class AudioExtElement extends HTMLElement {
     _prepareYoutube(element, src, cb) {
         let duration, begin;
 
-        const player = new YT.Player(element, {
-            width: 600,
-            height: 400,
-            videoId: src,
-            playerVars: {
-                playsinline: 1,
-                autoplay: 1,
-            },
-            events: {
-                onReady: () => {
-                    this.duration = player.getDuration();
-                    cb();
+        YT.ready(() => {
```

Comment:
nit. could we just promisify and await this instead of increasing the nested depth?

ie.
```
await new Promise((resolve) => YT.ready(resolve));
```